gfop/foodcounts.py

```diff
@@ -0,0 +1,296 @@
+
+
+class FoodCounts:
+    def __init__(
+        self,
+        gnps_network: str,
+        sample_types: str,
+        all_groups: List[str],
+        some_groups: List[str],
+        levels: int = 6,
+    ) -> None:
+        """
+        Initializes the FoodCounts object and automatically creates the food counts for all levels and all types.
+
+        Args:
+            gnps_network (str): Path to the TSV file generated from classical molecular networking.
+            sample_types (str): One of 'simple', 'complex', or 'all'.
+            all_groups (List[str]): List of study spectrum file groups.
+            some_groups (List[str]): List of reference spectrum file groups.
+            levels (int): Number of levels to calculate food counts for.
+        """
+        # Load GNPS network data
+        self.gnps_network = pd.read_csv(gnps_network, sep="\t")
+        self.sample_types = self._load_sample_types(sample_types)
+        self.all_groups = all_groups
+        self.some_groups = some_groups
+        self.levels = levels
+        self.food_metadata = self._load_food_metadata()
+
+        # Validate group names
+        self._validate_groups()
+
+        # Generate sample metadata and counts
+        self.sample_metadata = self._get_sample_metadata()
+        self.counts = self.create()
+
+    def _validate_groups(self) -> None:
+        """
+        Validates that the provided group names exist in the GNPS network data.
+        Raises a ValueError if invalid group names are found.
+        """
+        valid_groups = set(self.gnps_network["DefaultGroups"].unique())
+
+        # Check all_groups
+        invalid_all_groups = set(self.all_groups) - valid_groups
+        if invalid_all_groups:
+            raise ValueError(
+                f"The following groups in all_groups are invalid: {invalid_all_groups}"
+            )
+
+        # Check some_groups
+        invalid_some_groups = set(self.some_groups) - valid_groups
+        if invalid_some_groups:
+            raise ValueError(
+                f"The following groups in some_groups are invalid: {invalid_some_groups}"
+            )
+
+    def _load_food_metadata(self) -> pd.DataFrame:
+        """
+        Reads Global FoodOmics ontology and metadata.
+
+        Returns:
+            pd.DataFrame: A dataframe containing Global FoodOmics ontology and metadata.
+        """
+        # Use importlib.resources if possible; fall back to pkg_resources
+        try:
+            with resources.open_text(
+                "data", "foodomics_multiproject_metadata.txt"
+            ) as stream:
+                gfop_metadata = pd.read_csv(stream, sep="\t")
+        except (ModuleNotFoundError, ImportError):
+            stream = pkg_resources.resource_stream(
+                __name__, "data/foodomics_multiproject_metadata.txt"
+            )
+            gfop_metadata = pd.read_csv(stream, sep="\t")
+
+        # Remove trailing whitespace
+        gfop_metadata = gfop_metadata.apply(
```

Comment:
**todo:** The metadata is packaged with the code, so just modify the file that it is fully correct.